src/main.cpp

```diff
@@ -40,22 +41,27 @@ int main(int argc, char **argv) {
   ros::Rate update_rate(update_rate_hz);
   ros::Time ts = ros::Time::now();
   while (!shouldShutdown) {
-    forqueSensorHW.update();
+    if(!forqueSensorHW.update()) {
+      errorShutdown = true;
+      break;
+    }
     ros::Duration d = ros::Time::now() - ts;
     ts = ros::Time::now();
     cm.update(ts, d);
     update_rate.sleep();
   }
 
   // wait for a bit, so controller_manager can unload controllers
-  ros::Duration shutdownDuration(1.5);
-  ros::Time shutdownBegin = ros::Time::now();
-  while (ros::Time::now() - shutdownBegin < shutdownDuration) {
-    ros::Duration d = ros::Time::now() - ts;
-    ts = ros::Time::now();
-    cm.update(ts, d);
+  if(!errorShutdown) {
```

Comment:
Ah, misread this indentation and I thought it was inside the loop above. 

```suggestion
  if (!errorShutdown) {
```